# --- T2-COPYRIGHT-NOTE-BEGIN ---
# T2 SDE: architecture/powerpc64/package/*/1000-gelic-skb-alloc.patch
# Copyright (C) 2020 - 2024 The T2 SDE Project
# 
# This Copyright note is generated by scripts/Create-CopyPatch,
# more information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License version 2 as used by the T2 SDE.
# --- T2-COPYRIGHT-NOTE-END ---

Since some MM rework around 5.x the ps3 gelic ethernet driver quickly runs
out of memory to allocate. Tweak some defaults to avoid this as much as
possible.

--- linux-6.10/mm/page_alloc.c.vanilla	2024-08-15 18:25:26.514273117 +0200
+++ linux-6.10/mm/page_alloc.c	2024-08-15 18:26:44.068278120 +0200
@@ -269,9 +269,17 @@
 #endif
 };
 
+#ifdef CONFIG_PPC_PS3
+int min_free_kbytes = 2048;
+#else
 int min_free_kbytes = 1024;
+#endif
 int user_min_free_kbytes = -1;
+#ifdef CONFIG_PPC_PS3
+static int watermark_boost_factor __read_mostly;
+#else
 static int watermark_boost_factor __read_mostly = 15000;
+#endif
 static int watermark_scale_factor = 10;
 
 /* movable_zone is the "real" zone pages in ZONE_MOVABLE are taken from */
